addons/l10n_id_efaktur/models/efaktur_document.py

```diff
@@ -0,0 +1,299 @@
+            # To do so, we adjust the first line in order to achieve the correct total
+            if sales:
+                diff_dpp = idr.round(eTax['JUMLAH_DPP'] - sum(sale['DPP'] for sale in sales))
+                total_sales_ppn = idr.round(eTax['JUMLAH_PPN'] - sum(sale['PPN'] for sale in sales))
+                # We will add the differences to the first line for which adding the difference will not result in a negative value.
+                for sale in sales:
+                    if sale['DPP'] + diff_dpp >= 0 and sale['PPN'] + total_sales_ppn >= 0:
+                        sale['HARGA_TOTAL'] += diff_dpp
+                        sale['DPP'] += diff_dpp
+                        diff_dpp = 0
+                        sale['PPN'] += total_sales_ppn
+                        total_sales_ppn = 0
+                        break
+
+                # We couldn't adjust everything in a single line as their values is too low.
+                # So we will instead slit the adjustment in multiple lines.
+                if diff_dpp or total_sales_ppn:
+                    for sale in sales:
+                        # DPP
+                        sale_dpp = sale['DPP']
+                        sale["DPP"] = max(0, sale["DPP"] + diff_dpp)
+                        diff_dpp -= (sale["DPP"] - sale_dpp)
+                        sale['HARGA_TOTAL'] = sale["DPP"]
+                        # PPN
+                        sale_ppn = sale['PPN']
+                        sale["PPN"] = max(0, sale["PPN"] + total_sales_ppn)
+                        total_sales_ppn -= (sale["PPN"] - sale_ppn)
+
+            # Values now being corrected, we can format them for the CSV
+            for sale in sales:
+                sale.update({
+                    'HARGA_TOTAL': float_repr(sale['HARGA_TOTAL'], idr.decimal_places),
+                    'DPP': float_repr(sale['DPP'], idr.decimal_places),
+                    'PPN': float_repr(sale['PPN'], idr.decimal_places),
+                })
+
+            output_head += _csv_row(fk_values_list, delimiter)
+            for sale in sales:
+                of_values_list = ['OF'] + [str(sale[f]) for f in OF_HEAD_LIST[1:-2]] + ['0', '0']
+                output_head += _csv_row(of_values_list, delimiter)
+
+        return output_head
+
+    @api.depends('invoice_ids')
+    def _compute_name(self):
+        """ First compute will be done at creation, from a selection of invoice(s).
+        We still want to allow to rename the document to another name if it makes sense.
+        """
+        for doc in self:
+            sorted_invoices = doc.invoice_ids.sorted('name')
+            name = []
+            if sorted_invoices:
+                name.append(sorted_invoices[0].name)
+                if len(sorted_invoices) > 1:
+                    name.append(sorted_invoices[-1].name)
+            doc.name = "%s - Efaktur (%s)" % (fields.Date.context_today(doc).strftime("%Y%m%d"), "....".join(name))
```

Comment:
taking the first and the last? what if they don't share the same sequence prefix? (different journal or month?) 

(not wrong but, arguably, not correct either... but then it's only a default value, so... not sure we care)

Reply:
Was discussed with tsb. We didn't came up with a better solution, and as you said it's a default value to distinguish between them so it should be fine :)